src/Utilities/VariableTemplates/VariableTemplates.jl

```diff
@@ -374,4 +374,27 @@ Base.@propagate_inbounds function Base.getindex(
     end
 end
 
+export unroll_map
+
+"""
+    unroll_map(f, N::Int, args...)
+
+Unroll loop inside GPU kernel. See the
+`ntuple_model` example in the test suite:
+
+```julia
+up = ntuple(i->src.ntuple_model[i].scalar_model.x, N)
+unroll_map(N) do i
+    dst.ntuple_model[i].scalar_model.x = up[i]
+end
+```
+"""
+unroll_map(f, N::Int, args...) = unroll_map(f, Val(N), args...)
+@generated function unroll_map(f::F, ::Val{N}, args...) where {F, N}
+    quote
+        Base.@_inline_meta
+        Base.Cartesian.@nexprs $N i -> f(i, args...)
+    end
+end
```

Comment:
Same comment as below: https://github.com/CliMA/ClimateMachine.jl/pull/1437/files#r469604066 about not needing a generated function